docs/guide/front-end-helpers.md

```diff
@@ -0,0 +1,319 @@
+| -------- | ----------- |
+| `el` | a DOM element |
+| `className` | a string to be added to the `class` attribute |
+
+```javascript
+const myElement = document.querySelector('[data-my-element]');
+
+apos.util.removeClass(myElement, 'is-hidden');
+```
+
+### `sameSite(uri)`
+
+Returns `true` if the URI pass in matches the same website (same host and port) as the current page. This is used some HTTP utility methods.
+
+| Argument | What is it? |
+| -------- | ----------- |
+| `uri` | a valid URI |
+
+```javascript
+const targetUrl = 'https://some-website.rocks/api/gems'
+const siteMatches = apos.util.sameSite(targetUrl);
+```
+
+::: note
+There is also a `runPlayers` method on `apos.util`. That is run for us using `apos.util.onReadyAndRefresh` and runs all registered widget players. It is unlikely that it will need to be run in project-level code.
+:::
+
+## HTTP request methods
+
+These are all available in the browser on `apos.util`, e.g., `apos.util.addClass(el, 'is-active')`. They include wrappers for common browser APIs, classic DOM traversal methods, and Apostrophe-specific utilities.
+
+| Method | What is it? |
+| -------- | ----------- |
+| [`get`](#get-url-options-callback) | Send a `GET` request. |
+| [`post`](#post-url-options-callback) | Send a `POST` request. |
+| [`patch`](#patch-url-options-callback) | Send a `PATCH` request. |
+| [`put`](#put-url-options-callback) | Send a `PUT` request. |
+| [`delete`](#delete-url-options-callback) | Send a `DELETE` request. |
+| [`remote`](#remotemet-hod-url-options-callback) | The HTTP request method that powers other request methods. |
+| [`parseQuery`](#parsequery-query) | Parses a URL query string, returning an object of parameters. |
+| [`addQueryToUrl`](#addquerytourl-url-data) | Returns a URL with data object properties added as a query string. |
+
+### `get(url, options, callback)`
+
+Send a `GET` request. The response will be returned via a Promise unless a callback is included. Query string data may be in `options.qs`. You do NOT have to pass a callback unless you must support IE11 and do not otherwise have Promise support.
+
+| Argument | What is it? |
+| -------- | ----------- |
+| `url` | The path to a resource or service |
+| `options` | Request options. See [`apos.http.remote`](#remote-method-url-options-callback) for details. |
+| `callback` | An optional callback function receiving when not using Promises. Receives `error` and `result` arguments. |
+
+```javascript
+async function logArticles() {
+  let articles;
+
+  try {
+    articles = await apos.http.get('/api/v1/article');
+    console.info(articles);
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+logArticles();
+```
+
+### `post(url, options, callback)`
+
+Send a `POST` request. The response will be returned via a Promise unless a callback is included. `POST` body data should be in `options.body`. You do NOT have to pass a callback unless you must support IE11 and do not otherwise have Promise support.
```

Comment:
"`body` should be an object, containing properties to be passed as POST body data." Otherwise they might think they have to URL encode by hand.